File: vol1/ch7/LeetCode776.cpp
class Solution {
public:
    vector<TreeNode*> splitBST(TreeNode* root,int target) {
        if(root==NULL)
            return {NULL,NULL};
        else if (root->val<=target) {
            vector<TreeNode*> tmp=splitBST(root->right,target);
            root->right=tmp[0];
            return {root,tmp[1]};
        }
        else {
            vector<TreeNode*> tmp=splitBST(root->left,target);
            root->left=tmp[1];
            return {tmp[0],root};
        }
    }
};

File: vol1/ch7/LeetCode776.py
class Solution:
    def splitBST(self, root:Optional[TreeNode],target:int)->List[Optional[TreeNode]]:
        if root==None:
            return [None,None]
        elif root.val<=target:
            tmp=self.splitBST(root.right,target)
            root.right=tmp[0]
            return [root,tmp[1]]
        else:
            tmp=self.splitBST(root.left,target)
            root.left=tmp[1]
            return [tmp[0],root] 
